components/esp_lvgl_port/test_apps/simd/main/test_lv_fill_functionality.c

```diff
@@ -0,0 +1,340 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <malloc.h>
+#include <inttypes.h>
+#include "unity.h"
+#include "esp_log.h"
+#include "lvgl.h"
+#include "lv_fill_common.h"
+
+// ------------------------------------------------- Defines -----------------------------------------------------------
+
+#define DBG_PRINT_OUTPUT false
+#define CANARY_BYTES 4
+
+// ------------------------------------------------- Macros and Types --------------------------------------------------
+
+static const char *TAG_LV_FILL_FUNC = "LV Fill Functionality";
+
+static blend_params_t *s_blend_params = NULL;
+static test_area_t *s_area = NULL;
+static char test_msg_buff[128];
+
+// ------------------------------------------------ Static function headers --------------------------------------------
+
+/**
+ * @brief Generate all the functionality test combinations
+ *
+ * - generate functionality test combinations, based on the provided test_params struct
+ *
+ * @param[in] test_params Pointer to structure with test_parameters
+ */
+static void functionality_test_matrix(func_test_params_t *test_params);
+
+/**
+ * @brief ARGB8888 functionality test case entry
+ *
+ * - initialize the ARGB8888 functionality test case
+ */
+static void argb8888_functionality(void);
+
+/**
+ * @brief RGB565 functionality test case entry
+ *
+ * - initialize the RGB565 functionality test case
+ */
+static void rgb565_functionality(void);
+
+/**
+ * @brief The actual LV Fill ARGB8888 functionality test
+ */
+static void lv_fill_argb8888_functionality(int w, int h, int stride, int unalign_byte);
+
+/**
+ * @brief The actual LV Fill RGB565 functionality test
+ */
+static void lv_fill_rgb565_functionality(int w, int h, int stride, int unalign_byte);
+
+/**
+ * @brief Run the LVGL blend API
+ */
+static void run_lv_blend_api(void *dest_buff_ansi, void *dest_buff_asm);
+
+// ------------------------------------------------ Test cases ---------------------------------------------------------
+
+/*
+Functionality tests
+
+Purpose:
+    - Test that an assembly version of LVGL blending API works correctly
+
+Procedure:
+    - Initialize structures needed for LVGL blending API
+    - Initialize input parameters (test array length, width, allocate array...) of the benchmark test
+    - Prepare testing matrix, to cover all the possible combinations of destination array widths, lengths, memory alignment...
+    - Run assembly version of the LVGL blending API
+    - Run ANSI C version of the LVGL blending API
+    - Compare the results
+    - Repeat above 3 steps for each test matrix setup
+    - Free structures needed for LVGL blending API
+*/
+
+TEST_CASE_MULTIPLE_STAGES("LV Fill functionality", "[lv_fill][functionality]",
```

Comment:
Could you explain why multi-stage test is used here? From docs, it seems that we might not need this https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/unit-tests.html#multi-stage-test-cases

Reply:
I was not aware of this usage of multi-stage tests. Just wanted to group functionality and performance tests into 2 separate groups. Will correct it. Thx for the docs reference.